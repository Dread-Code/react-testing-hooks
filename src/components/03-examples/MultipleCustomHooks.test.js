import React from "react";
import { shallow } from "enzyme/build";
import { MultipleCustomHooks } from "./MultipleCustomHooks";
import { useFetch } from "../../hooks/useFetch";
import "@testing-library/jest-dom";
import { useCouter } from "../../hooks/useCouter";
/**
 * Jest.mock is a function that break up
 * the functions that change the state
 * of our component, yo have to rememeber the
 * unit testing should be of our actual
 * component and divide the external
 * function from it
 */

jest.mock("../../hooks/useFetch");
jest.mock("../../hooks/useCouter");

describe("MultipleCustomHooks", () => {
  /**
   * With this we return a "fake" data from
   * the function that we mocked
   */
  useCouter.mockReturnValue({
    counter: 10,
    increment: () => {},
  });

  test("should show corectly", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match the information", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Lucas",
          quote: "Hola Putos",
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show the information correctly", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Lucas",
          quote: "Hola Putos",
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-0").text().trim()).toBe("Hola Putos");
    expect(wrapper.find("footer").text().trim()).toBe("Lucas");
  });
});
