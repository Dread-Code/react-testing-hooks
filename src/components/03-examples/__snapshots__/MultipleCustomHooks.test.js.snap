// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleCustomHooks should match the information 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    >
       
      Hola Putos
       
    </p>
    <footer
      className="blockquote-footer"
    >
       
      Lucas
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;

exports[`MultipleCustomHooks should show corectly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</div>
`;
